metric-publishers/emf-metric-publisher/src/main/java/software/amazon/awssdk/metrics/publishers/emf/MetricEmfConverter.java

```diff
@@ -0,0 +1,368 @@
+
+        // Write Dimensions array
+        jsonWriter.writeFieldName("Dimensions");
+        jsonWriter.writeStartArray();
+        jsonWriter.writeStartArray();
+        for (String dimension : realDimensionStrings) {
+            jsonWriter.writeValue(dimension);
+        }
+        jsonWriter.writeEndArray();
+        jsonWriter.writeEndArray();
+
+        // Write Metrics array
+        jsonWriter.writeFieldName("Metrics");
+        jsonWriter.writeStartArray();
+
+
+        // Write metric definitions
+        for (String metricName : metricNames) {
+            jsonWriter.writeStartObject();
+            jsonWriter.writeFieldName("Name");
+            jsonWriter.writeValue(metricName);
+
+            // Add Unit if available
+            if (hasUnit(metricName)) {
+                jsonWriter.writeFieldName("Unit");
+                jsonWriter.writeValue(getMetricUnit(metricName));
+            }
+
+            jsonWriter.writeEndObject();
+        }
+
+        jsonWriter.writeEndArray(); // End Metrics array
+        jsonWriter.writeEndObject(); // End CloudWatchMetrics object
+        jsonWriter.writeEndArray(); // End CloudWatchMetrics array
+        jsonWriter.writeEndObject(); // End _aws object
+    }
+
+    private void writeMetricValues(JsonWriter jsonWriter, Map<String, List<Object>> metrics) {
+        for (Map.Entry<String, List<Object>> entry : metrics.entrySet()) {
+            String metricName = entry.getKey();
+            List<Object> values = entry.getValue();
+
+            // For dimension metrics, write the last value
+            if (isDimension(metricName)) {
+                jsonWriter.writeFieldName(metricName);
+                jsonWriter.writeValue((String) values.get(values.size() - 1));
+            } else {
+                //skip string values
+                if (values.get(0) instanceof String) {
+                    continue;
+                }
+                // For regular metrics, if there's only one value, write it directly
+                if (values.size() == 1) {
+                    jsonWriter.writeFieldName(metricName);
+                    writeProcessedValue(jsonWriter, processValue(values.get(0)));
+                } else {
+                    // If there are multiple values, write as an array
+                    jsonWriter.writeFieldName(metricName);
+                    jsonWriter.writeStartArray();
+                    for (Object value : values) {
+                        writeProcessedValue(jsonWriter, processValue(value));
+                    }
+                    jsonWriter.writeEndArray();
+                }
+            }
+        }
+    }
+
+
+    private boolean isDimension(String metricName) {
+        return metricName != null && config.getDimensionStrings().contains(metricName);
+    }
+
+    private boolean hasUnit(String metricName) {
+        // Implement logic to determine if metric should have a unit
+        return metricName.contains("Duration");
```

Comment:
Hmm, should we use type to determine instead? We have other numeric metrics that doesn't contain `Duration` such as `PendingConcurrencyAcquires`